python_modules/dagster/dagster/_daemon/monitoring/monitoring_daemon.py

```diff
@@ -3,35 +3,43 @@
 import time
 from typing import Iterator, Optional
 
+import pendulum
+
 from dagster import (
     DagsterInstance,
     _check as check,
 )
-from dagster._core.events import DagsterEventType
+from dagster._core.events import DagsterEventType, EngineEventData
 from dagster._core.launcher import WorkerStatus
 from dagster._core.storage.pipeline_run import (
     IN_PROGRESS_RUN_STATUSES,
-    DagsterRun,
     DagsterRunStatus,
+    RunRecord,
     RunsFilter,
 )
+from dagster._core.storage.tags import MAX_RUNTIME_TAG
 from dagster._core.workspace.context import IWorkspace, IWorkspaceProcessContext
 from dagster._utils import DebugCrashFlags
 from dagster._utils.error import SerializableErrorInfo, serializable_error_info_from_exc_info
 
+DEFAULT_MAX_RUNTIME = 60 * 60 * 12
```

Comment:
what is this? I don't think we should have a default limit

Reply:
https://github.com/dagster-io/dagster/pull/12920#discussion_r1134645178 @gibsondan disagrees

Reply:
I think it could be reasonable to offer that as an optional configuration alongside the start timeout https://github.com/dagster-io/dagster/blob/master/helm/dagster/values.yaml#L1095 though I think it will either be very long or cause unintended interrupts. I definitely don't think we should have a hard coded max run time for all jobs unless you specifically override it with a tag. Users should be able to opt in one job to a timeout without affecting others.

Reply:
Maybe we just leave the default bit out for now - and address later based on actual feedback, because it's unclear to me how the tag ergonomics would work there